src/sparse_blas/backends/mkl_common/mkl_helper.hpp

```diff
@@ -1,56 +1,102 @@
-/*******************************************************************************
-* Copyright 2023 Codeplay Software Ltd.
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
 *
-* Licensed under the Apache License, Version 2.0 (the "License");
-* you may not use this file except in compliance with the License.
-* You may obtain a copy of the License at
+*      http://www.apache.org/licenses/LICENSE-2.0
 *
-* http://www.apache.org/licenses/LICENSE-2.0
+*  For your convenience, a copy of the License has been included in this
+*  repository.
 *
-* Unless required by applicable law or agreed to in writing,
-* software distributed under the License is distributed on an "AS IS" BASIS,
-* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-* See the License for the specific language governing permissions
-* and limitations under the License.
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
 *
-*
-* SPDX-License-Identifier: Apache-2.0
-*******************************************************************************/
+**************************************************************************/
 
-// MKLCPU and MKLGPU backends include
-// This include defines its own oneapi::mkl::sparse namespace with some of the types that are used here: matrix_handle_t, index_base, transpose, uolo, diag.
-#include <oneapi/mkl/spblas.hpp>
+#ifndef _ONEMKL_SRC_SPARSE_BLAS_BACKENDS_MKL_COMMON_MKL_HELPER_HPP_
+#define _ONEMKL_SRC_SPARSE_BLAS_BACKENDS_MKL_COMMON_MKL_HELPER_HPP_
 
-// Includes are set up so that oneapi::mkl::sparse namespace refers to the MKLCPU and MKLGPU backends namespace (oneMKL product)
-// in this file.
-// oneapi::mkl::sparse::detail namespace refers to the oneMKL interface namespace.
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
 
+#include "oneapi/mkl/exceptions.hpp"
```

Comment:
Is it that when you use `<oneapi/mkl/exceptions.hpp>`, it uses Intel oneMKL's include file, and when you use `"oneapi/mkl/exceptions.hpp"`, it uses oneMKL Interfaces' header?

Reply:
Yes, the quotes are used to include files from the same project by convention. It can be a bit more difficult than that to include files from the oneMKL Intel Product due to how the include paths are set up. We've add difficult issues with that for the DFT domain. Here I only need to include the oneMKL SYCL Interface headers anyway.